numpy/core/src/umath/ufunc_object.c

```diff
@@ -2260,7 +2317,7 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
      */
     iter_ndim = broadcast_ndim;
     for (i = nin; i < nop; ++i) {
-        iter_ndim += ufunc->core_num_dims[i];
+        iter_ndim += keepdims ? 1 : ufunc->core_num_dims[i];
```

Comment:
Why is `keepdims` being added in the loop? That seems wrong to me

Reply:
It does need to be there, so that the iterator can allocate dimensions in the correct order (see comment just above this piece of code)